tests/virt_autotest/virt_utils.pm

```diff
@@ -287,6 +288,10 @@ sub lpar_cmd {
     }
     unless ($args->{ignore_return_code} || !$ret) {
         record_info('INFO', "Command $cmd run on S390X LPAR: FAIL");
+        #Figure out new failure here, so collect and upload
+        #supportconfig log from S390X LPAR as required
+        upload_system_log::upload_supportconfig_log();
+        script_run "rm -rf scc_*";
```

Comment:
Above 4 lines are not in the base class post_fail_hook? We discussed this and agreed to put there, right? Difference is lpar_cmd is just a lib function to execute a command on lpar, but it does not mean the end of test, and it can be soft-failed to continue test. So better to be put in post_fail_hook. You can take similar way with other places, if s390_arch, then call this upload_supportconfig. 
Others LGTM.

Reply:
Great, had move upload supportconfig to virt_autotest_base based class now. for now,  when s390x GI | GU Test fail, will call  virt_autotest_base::post_fail_hook() dun to collect and upload supportconfig log from s390x lpar.